v2/tools/generator/internal/functions/kubernetes_admissions_validations.go

```diff
@@ -77,3 +85,59 @@ func validateResourceReferencesBody(codeGenerationContext *astmodel.CodeGenerati
 
 	return body
 }
+
+func validateImmutablePropertiesFunction(resourceFn *ResourceFunction, codeGenerationContext *astmodel.CodeGenerationContext, receiver astmodel.TypeName, methodName string) *dst.FuncDecl {
+
+	receiverIdent := resourceFn.IdFactory().CreateReceiver(receiver.Name())
+	receiverType := receiver.AsType(codeGenerationContext)
+
+	runtimePackage := codeGenerationContext.MustGetImportedPackageName(astmodel.APIMachineryRuntimeReference)
+
+	fn := &astbuilder.FuncDetails{
+		Name:          methodName,
+		ReceiverIdent: receiverIdent,
+		ReceiverType: &dst.StarExpr{
+			X: receiverType,
+		},
```

Comment:
There's a helper function in `astbuilder` for this.

Reply:
Updated to use `astbuilder.Deference`